### ISSUE TITLE
Huge API calls to `iam.amazonaws.com` 

### ISSUE THREAD

**Issue by haooliveira84** [crossplane-contrib/provider-aws#2075]
### What happened?
We're observed the increase API calls to `iam.amazonaws.com` with event `UpdateAssumeRolePolicy`

### How can we reproduce it?

Provisioning the IAM Role, IAM Attach Policy and a IAM Policy.

### What environment did it happen in?
Crossplane version: 
Crossplane-UXP: v1.14.7-up.1
Crossplane-provider version: v0.47.2

### Part of CloudTrail event

```json
...
    "eventSource": "iam.amazonaws.com",
    "eventName": "UpdateAssumeRolePolicy",
    "awsRegion": "us-east-1",
    "userAgent": "aws-sdk-go-v2/1.19.0 os/linux lang/go#1.21.2 md/GOOS#linux md/GOARCH#amd64 api/iam#1.14.0 crossplane-provider-aws/v0.47.2",
    "requestParameters": {
        "roleName": "role-application",
        "policyDocument": "{\n  \"Version\": \"2012-10-17\",\n  \"Statement\":\n    [\n      {\n        \"Effect\": \"Allow\",\n        \"Principal\": {\n          \"Service\": [\n            \"apigateway.amazonaws.com\",\n            \"lambda.amazonaws.com\",\n            \"states.amazonaws.com\"\n          ]\n        },\n        \"Action\": [\n          \"sts:AssumeRole\"\n        ]\n      }\n    ]\n}\n"
    },
...
```

**Comment by haarchri** [comment]
This behaviour is fixed in v0.48.0 ref: https://github.com/crossplane-contrib/provider-aws/pull/2055 and i tested one: https://github.com/crossplane-contrib/provider-aws/compare/master...haarchri:provider-aws:test/iam-policy-compare?expand=1

**Comment by MisterMX** [comment]
Closing, since this seems to be fixed.
